File: Append and Delete.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the appendAndDelete function below.
def appendAndDelete(s, t, k):
    i=0
    n1=int(len(s))
    n2=int(len(t))
    min_len=min(n1,n2)
    for i in range(min_len):
        if(s[i]!=t[i]):
            break
    count=n1+n2-2*i
    if(k==count or k>=n1+n2) or (count%2==k%2 and count<=k):
        return "Yes"
    else:
        return "No"
   



if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    t = input()

    k = int(input())

    result = appendAndDelete(s, t, k)

    fptr.write(result + '\n')

    fptr.close()

File: Append and Delete.cpp
#include <bits/stdc++.h>

using namespace std;

// Complete the appendAndDelete function below.
string appendAndDelete(string s, string t, int k) {
    int i=0;
    int n1=s.size();
    int n2=t.size();
    int min_len=min(n1,n2);
    for(i=0;i<min_len;i++){
        if(s[i]!=t[i]);
        break;
    }
    int count=n1+n2-2*i;
    if((k==count || k>=n1+n2) || (count%2==k%2 && count<=k)){
        return "Yes";
    }
    else{
        return "No";
    }
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string s;
    getline(cin, s);

    string t;
    getline(cin, t);

    int k;
    cin >> k;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');

    string result = appendAndDelete(s, t, k);

    fout << result << "\n";

    fout.close();

    return 0;
}
